Client/game_sa/CEntitySA.cpp

```diff
@@ -16,6 +16,39 @@ extern CGameSA* pGame;
 unsigned long CEntitySA::FUNC_CClumpModelInfo__GetFrameFromId;
 unsigned long CEntitySA::FUNC_RwFrameGetLTM;
 
+void CEntitySAInterface::TransformFromObjectSpace(CVector& outPosn, CVector const& offset)
+{
+    ((void(__thiscall*)(CEntitySAInterface*, CVector&, CVector const&))0x533560)(this, outPosn, offset);
+}
+
+CVector* CEntitySAInterface::GetBoundCentre(CVector* pOutCentre)
+{
+    return ((CVector * (__thiscall*)(CEntitySAInterface*, CVector*))0x534250)(this, pOutCentre);
+}
+
+CRect* CEntitySAInterface::GetBoundRect_(CRect* pRect)
+{
+    CColModelSAInterface* colModel = CModelInfoSAInterface::GetModelInfo(m_nModelIndex)->pColModel;
+    CVector               vecMin = colModel->boundingBox.vecMin;
+    CVector               vecMax = colModel->boundingBox.vecMax;
+    CRect                 rect;
+    CVector               point;
+    TransformFromObjectSpace(point, vecMin);
+    rect.StretchToPoint(point.fX, point.fY);
+    TransformFromObjectSpace(point, vecMax);
+    rect.StretchToPoint(point.fX, point.fY);
+    float maxX = vecMax.fX;
+    vecMax.fX = vecMin.fX;
+    vecMin.fX = maxX;
+    TransformFromObjectSpace(point, vecMin);
+    rect.StretchToPoint(point.fX, point.fY);
+    TransformFromObjectSpace(point, vecMax);
+    rect.StretchToPoint(point.fX, point.fY);
+    *pRect = rect;
+    pRect->FixIncorrectTopLeft();            // Fix custom map collision crashes in CPhysical class (infinite loop)
```

Comment:
Nit:

```suggestion
    pRect->FixIncorrectTopLeft();            // Fix #1613 custom map collision crashes in CPhysical class (infinite loop)
```


Essentially, it would be good for a future reader to know why this line was added. Since we don't have an issue out for this bug, linking to this PR (#1613) since it has a sufficient enough description.

If we incorporate https://github.com/multitheftauto/mtasa-blue/pull/1613/files/7e70ebb8a0d2ca68b0e4ee7dc3b509413e92485f#diff-5699c2fa9412f449313c431dfc9b4e83 then we could perhaps just include a comment in the function definition (and perhaps a hint "read ThisFuncName for more information on why this is necessary").